R/tmb.R

```diff
@@ -514,20 +524,9 @@ fit_mmrm <- function(formula,
     tmb_opt$objective <- tmb_opt$value
     tmb_opt$value <- NULL
   }
-  fit <- h_mmrm_tmb_fit(tmb_object, tmb_opt, data, weights, formula_parts, tmb_data)
+  fit <- h_mmrm_tmb_fit(tmb_object, tmb_opt, formula_parts, tmb_data)
   h_mmrm_tmb_check_conv(tmb_opt, fit)
-
-  fun_call <- match.call()
-  fun_call$formula <- eval(formula_parts$formula)
-  fun_call$data <- ifelse(
-    !is.null(attr(data, which = "dataname")),
-    attr(data, which = "dataname"),
-    toString(match.call()$data)
-  )
-  weights_str <- attr(weights, which = "dataname")
-  if (!is.null(weights_str)) {
-    fun_call$weights <- weights_str
-  }
-  fit$call <- fun_call
+  fit$call <- match.call()
+  fit$call$formula <- formula_parts$formula
```

Comment:
for the data, use the symbol directly; in `mmrm` the call is also retrieved. The reason to have two locations of call is that `fit_mmrm` is still exported so we need to use it.  For weights, I am just wondering if a "string" indicating the column in the data should be better than providing a numeric value

Reply:
hm interesting. I remember we had some problem with the data thing before. but I guess as long as the tests pass this should be fine.

and then the same logic as for data applies to the weights thing?